modules/oe_theme_helper/src/Plugin/Field/FieldFormatter/FeaturedMediaRenderedEntityFormatter.php

```diff
@@ -0,0 +1,71 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\oe_theme_helper\Plugin\Field\FieldFormatter;
+
+use Drupal\Core\Field\FieldItemInterface;
+use Drupal\Core\Field\FieldItemListInterface;
+use Drupal\Core\Field\Plugin\Field\FieldFormatter\EntityReferenceFormatterBase;
+use Drupal\file\Entity\File;
+
+/**
+ * Plugin implementation of the 'Featured media as entity' formatter.
+ *
+ * @FieldFormatter(
+ *   id = "oe_featured_media_entity_view",
+ *   label = @Translation("Rendered entity"),
+ *   description = @Translation("Display the entity of the referenced media and the caption."),
+ *   field_types = {
+ *     "oe_featured_media"
+ *   }
+ * )
+ */
+class FeaturedMediaRenderedEntityFormatter extends EntityReferenceFormatterBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function viewElements(FieldItemListInterface $items, $langcode) {
+    $elements = [];
+
+    foreach ($items as $delta => $item) {
+      $elements[$delta] = $this->viewElement($item);
+    }
+
+    return $elements;
+  }
+
+  /**
+   * Renders a single field item.
+   *
+   * @param \Drupal\Core\Field\FieldItemInterface $item
+   *   The individual field item.
+   *
+   * @return array
+   *   The ECL media-container parameters.
+   */
+  protected function viewElement(FieldItemInterface $item): array {
+    $element = [];
+    $element['_front_params']['description'] = $item->caption;
+
+    switch ($item->entity->bundle()) {
+      case 'remote_video':
+        $entity = $item->entity;
```

Comment:
From my point of view I would prefer to check whether $entity exists or not because it can be removed outside.
See how we did it already - https://github.com/openeuropa/oe_theme/pull/513/files#diff-175fc9d25ec415590a71139896c6b64dR32.

Reply:
done as in the link